src/app/components/courses-page/courses-page.component.ts

```diff
@@ -1,10 +1,31 @@
-import { Component } from '@angular/core';
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+
+import { courses } from 'src/app/shared/mock/courses';
+import { ICourse } from 'src/app/shared/interfaces/course';
 
 @Component({
   selector: 'app-courses-page',
   templateUrl: './courses-page.component.html',
-  styleUrls: ['./courses-page.component.scss']
+  styleUrls: ['./courses-page.component.scss'],
 })
-export class CoursesPageComponent {
+export class CoursesPageComponent implements OnInit, OnChanges {
+  coursesArray: ICourse[] = [];
+
+  constructor() {
+    console.log('Constructor');
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    console.log('ngOnChanges');
+    console.log(changes);
+  }
+
+  ngOnInit(): void {
+    console.log('ngOnInit');
+    this.coursesArray = courses;
+  }
 
-}
+  onDeleteCourseItem(id: number | string | undefined): void {
```

Comment:
Number | string type, if you mean number can lead to potential errors, it is better to avoid such cases
And I guess there is no need to write type _undefined_ here, you may just add the check _id_ is present further in the method

Reply:
ok